File: po/lista3/zad1.cs
/*
 * Katarzyna Trzos
 * lista 3 zad 1 listy
 * mcs -t:library  lista.cs
 * mcs zad1.cs -r:lista.dll
 * mono zad1.exe
 */
using System;

class Program
{
    public static void Main()
    {
        List<int> L  = new List<int>();
        L.push_back(2);
        L.push_back(3);
        L.push_front(1);
        Console.WriteLine(L.pop_back());
        Console.WriteLine(L.pop_back());
        Console.WriteLine(L.pop_back());
        L.push_back(2);
        L.push_back(3);
        L.push_front(1);
        Console.WriteLine(L.pop_front());
        Console.WriteLine(L.pop_front());
        Console.WriteLine(L.pop_front());
        Console.WriteLine(L.pop_back());
    }
}

File: wdi/lista8/zad1.c
#include <stdio.h>

int t1(int n)
{
    if (n == 1)
        return 1;
    else
    {
        return t1(n - 1) + n;
    }
}

int t2(int n)
{
    if (n == 1)
        return 0;
    if (n % 2 == 0)
    {
        return t2(n / 2) + 1;
    }
    else
    
        return t2((n+1) / 2) + 1;
        return t2((n + 1) / 2) + 1;
}

int t3(int n)
{
    if (n == 1)
        return 1;
    if (n % 2 == 0)
    {
        return 2 * t3(n / 2) + 1;
    }
    return t3((n - 1) / 2) + 1 + t3((n + 1) / 2) + 1;
}

int main()
{

    printf("zad 1  T(16)=%d\n", t1(16));
    printf("zad 2  T(16)=%d\n", t2(16));
    printf("zad 3  T(16)=%d\n", t2(16));
}